Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -0,0 +1,213 @@
+        
+        // Extract Edge Network environment level from shell environment
+        if let environment = EdgeEnvironment() {
+            self.edgeEnvironment = environment
+        }
+        print("Using Edge Network environment: \(edgeEnvironment.rawValue)")
+        // Extract Edge location hint from shell environment
+        if let locationHint = EdgeLocationHint() {
+            self.edgeLocationHint = locationHint
+        }
+        
+        let waitForRegistration = CountDownLatch(1)
+        MobileCore.setLogLevel(.trace)
+        MobileCore.configureWith(appId: "94f571f308d5/6b1be84da76a/launch-023a1b64f561-development")
+        MobileCore.registerExtensions([Identity.self, Edge.self], {
+            print("Extensions registration is complete")
+            waitForRegistration.countDown()
+        })
+        XCTAssertEqual(DispatchTimeoutResult.success, waitForRegistration.await(timeout: 2))
+        
+        // Set Edge location hint value if one is set for the test
+        if edgeLocationHint != nil {
+            print("Setting Edge location hint to: \(String(describing: edgeLocationHint?.rawValue))")
+            Edge.setLocationHint(edgeLocationHint?.rawValue)
+        }
+        else {
+            print("No preset Edge location hint is being used for this test.")
+        }
+    }
+    
+    public override func tearDown() {
+        super.tearDown()
+
+        // to revisit when AMSDK-10169 is available
+        // wait .2 seconds in case there are unexpected events that were in the dispatch process during cleanup
+        usleep(200000)
+        EventHub.reset()
+        UserDefaults.clearAll()
+        FileManager.default.clearCache()
+    }
+    
+    // TODO: create specific listeners for type: Edge + source: * (wildcard) and capture all the response handles for the test event
+    // TODO: create specific listeners for error responses
+    func registerEdgeLocationHintListener(_ listener: @escaping EventListener) {
+        MobileCore.registerEventListener(type: FunctionalTestConst.EventType.EDGE, source: "locationHint:result", listener: listener)
+    }
+    
+    /// Converts a JSON string into the provided type.
+    ///
+    /// NOTE: caller is reponsible for providing the correct casting type resulting JSON, otherwise decoding will fail
+    func convertToJSON<T>(_ jsonString: String) -> T? {
+        guard let jsonData = jsonString.data(using: .utf8) else {
+            XCTFail("Unable to convert provided JSON string to Data: \(jsonString)")
+            return nil
+        }
+        
+        guard let jsonDictionary = try? JSONSerialization.jsonObject(with: jsonData, options: []) as? T else {
+            XCTFail("Unable to convert provided JSON string to JSON type \(T.self)")
+            return nil
+        }
+        return jsonDictionary
+    }
+
+    // MARK: sample tests for the FunctionalTest framework usage
+    func testSample_AssertUnexpectedEvents() {
```

Comment:
title and surrounding comments are misleading - they seem copied from functional tests?